tests/test_functions.py

```
@@ -0,0 +1,137 @@
+from itertools import permutations, product
```

Ruff (0.11.9)

`itertools.permutations` imported but unused

Remove unused import: `itertools.permutations`

(F401)